src/vs/workbench/contrib/tasks/common/problemMatcher.ts

```diff
@@ -1284,7 +1284,7 @@ class ProblemPatternRegistryImpl implements IProblemPatternRegistry {
 
 	private fillDefaults(): void {
 		this.add('msCompile', {
-			regexp: /^(?:\s+\d+>)?(\S.*)\((\d+|\d+,\d+|\d+,\d+,\d+,\d+)\)\s*:\s+(error|warning|info)\s+(\w+\d+)\s*:\s*(.*)$/,
+			regexp: /^(?:\s*\d+>)?(\S.*)\((\d+|\d+,\d+|\d+,\d+,\d+,\d+)\)\s*:\s+(fatal ? error|error|warning|info)\s+(\w+\d+)\s*:\s*(.*)$/,
```

Comment:
```suggestion
			regexp: /^(?:\s*\d+>)?(\S.*)\((\d+|\d+,\d+|\d+,\d+,\d+,\d+)\)\s*:\s+((fatal )?error|warning|info)\s+(\w+\d+)\s*:\s*(.*)$/,
```

Reply:
Confusingly the ` ?` is actually functional. When compiling directly with CL the output will be `fatal error`. When compiling through MSBuild the output will be `fatal  error` with an extra space.

We can combine the `fatal error` and `error` parts, but the extra space will still need to be handled. I think it would be:
```
regexp: /^(?:\s*\d+>)?(\S.*)\((\d+|\d+,\d+|\d+,\d+,\d+,\d+)\)\s*:\s+((fatal ? )?error|warning|info)\s+(\w+\d+)\s*:\s*(.*)$/,
```
https://regex101.com/r/aENBTR/1

Reply:
yep, forgot the extra space. thanks